OpenKeychain/src/main/java/org/sufficientlysecure/keychain/receiver/NetworkReceiver.java

```diff
@@ -0,0 +1,51 @@
+package org.sufficientlysecure.keychain.receiver;
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import org.sufficientlysecure.keychain.Constants;
+import org.sufficientlysecure.keychain.util.Log;
+
+public class NetworkReceiver extends BroadcastReceiver {
+
+    Context context;
+    public NetworkReceiver(){}
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        ConnectivityManager conn = (ConnectivityManager)
+                context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = conn.getActiveNetworkInfo();
+
+        if (networkInfo != null && networkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
+
+            // broadcaster receiver disabled
+            setWifiReceiverComponent(false, context);
+            //KeyserverSyncAdapterService.enableKeyserverSync(context);
```

Comment:
`enableKeyserverSync` is meant to enable the periodic 3 day sync. Take a look at `SYNC_NOW` in `KeyserverSyncAdapterService` for a more deterministic way of starting a sync (start the service with that intent).